packages/auth/package.json

```diff
@@ -22,19 +22,33 @@
     "@skillrecordings/cookies": "workspace:*",
     "axios": "^0.21.1",
     "client-oauth2": "^4.3.3",
+    "jsdom": "^17.0.0",
+    "msw": "^0.34.0",
     "next": "11.1.0"
   },
   "devDependencies": {
     "@skillrecordings/scripts": "workspace:*",
     "@skillrecordings/tsconfig": "workspace:*",
     "@skillrecordings/types": "workspace:*",
+    "@types/jsdom": "^16.2.13",
+    "jest-localstorage-mock": "^2.4.14",
     "typescript": "^4.2.4"
   },
   "publishConfig": {
     "access": "public"
   },
   "jest": {
-    "preset": "@skillrecordings/scripts/jest/node"
+    "preset": "@skillrecordings/scripts/jest/node",
+    "resetMocks": false,
+    "setupFiles": [
+      "jest-localstorage-mock"
+    ],
+    "passWithNoTests": true,
+    "testEnvironment": "jsdom",
+    "setupFilesAfterEnv": [
+      "<rootDir>/setupTests.js"
+    ],
+    "testURL": "http://localhost:3000"
```

Comment:
the `tesURL` **absolutely needs to be set** or you can't change the URL later. Once it is set, you can use `window.history.pushState` to change the current `href`. This was a whole thing as far as I can tell, with a lot of sad times and verbose solutions.

here's the comment that did the trick:

https://github.com/facebook/jest/issues/890#issuecomment-506926841

if you see a better solution, please let me know 😅